### DISCUSSION TITLE
Using non-standard methods (and URIs)

### DISCUSSION THREAD

**Discussion by postlund** [aio-libs/aiohttp#5691] [Q&A]
I would like to use aiohttp to implement an RTSP server (for RAOP in my case). It's very similar to HTTP, but uses other methods and allows a URI as path. A typical request might look like this:

```raw
SETUP rtsp://10.0.10.254/1085946124 RTSP/1.0
CSeq: 2
User-Agent: AirPlay/540.31
DACP-ID: A851074254310A45
Active-Remote: 4019753970
Client-Instance: A851074254310A45
Transport: RTP/AVP/UDP;unicast;interleaved=0-1;mode=record;control_port=55433;timing_port=55081
```

Is it possible to add routes for these kinds of methods (and URIs), or will I have to use some other library instead?

**Comment by Dreamsorcerer** [comment]
Don't see why you wouldn't be able to: https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.UrlDispatcher.add_route

The add_get() etc. methods are just shortcuts as mentioned in the documentation.

**Reply by Dreamsorcerer** [reply]
Actually, the URI as path might not be trivial, I'd suggest trying it out and reporting back any problems.

**Reply by postlund** [reply]
I actually tried `add_route` and I can call it just fine, but it still doesn't work. I get an error about invalid method.
